subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -490,10 +492,32 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 			}
 			break;
 		case 'k':
-			params->security = atoi(optarg);
+			params->security = BIT(WIFI_SECURITY_TYPE_PSK);
+			params->mfp = WIFI_MFP_OPTIONAL;
+
+			/* Security type (optional) */
+
+			security = (char *)k_malloc(strlen(optarg));
+
+			if (!security) {
+				PR_ERROR("Failed to allocate memory for security config\n");
+				return -ENOMEM;
+			}
+
+			memset(security, 0, strlen(optarg));
+			strcpy(security, optarg);
```

Comment:
do we use strcpy or strncpy? Can you kindly please check this

Reply:
Used strcpy here since size of `security` buffer is equal to length of optarg string. According to another suggestion that I got from Robert Lubos, will make changes to avoid this completely.